src/meta/server_state.cpp

```diff
@@ -3269,8 +3269,373 @@ void server_state::set_max_replica_count(configuration_set_max_replica_count_rpc
              response.old_max_replica_count,
              new_max_replica_count);
 
-    // TODO: update partition-level and app-level max_replica_count
-    response.err = ERR_OK;
+    // update max_replica_count of each partition from index 0
+    tasking::enqueue(LPC_META_STATE_HIGH,
+                     tracker(),
+                     [this, rpc]() { update_partition_max_replica_count(0, rpc); },
+                     server_state::sStateHash);
+}
+
+// ThreadPool: THREAD_POOL_META_STATE
+void server_state::update_partition_max_replica_count(int32_t partition_index,
+                                                      configuration_set_max_replica_count_rpc rpc)
+{
+    const auto &app_name = rpc.request().app_name;
+    auto &response = rpc.response();
+
+    zauto_write_lock l(_lock);
+
+    auto app = get_app_and_check_exist(app_name, response);
+    if (app == nullptr) {
+        response.hint_message +=
+            fmt::format(" while updating max_replica_count of partition {}", partition_index);
+        derror_f("failed to set max_replica_count: app_name={}, error_code={}, hint_message={}",
+                 app_name,
+                 response.err.to_string(),
+                 response.hint_message);
+        return;
+    }
+
+    const auto app_id = app->app_id;
+
+    dassert_f(app->is_stateful,
```

Comment:
if not `main business logic`, such as `read/write`, not suggest execute dassert, which will cause unnecessary downtime of the cluster. You just need to return error to disable this operation

Reply:
> if not `main business logic`, such as `read/write`, not suggest execute dassert, which will cause unnecessary downtime of the cluster. You just need to return error to disable this operation

Actually `dassert(app->is_stateful)` is put here according to existing code,  such as 
 L1860, `server_state::on_update_configuration`. I think this assertion is just for the stateful table, is it ? On the other hand, it seems that for now there's no stateless table ?

Reply:
Yeah,  there's no stateless table. But`server_state::on_update_configuration` is `main logic`, it must make sure current state is correct, so it `dassert` is acceptable. This added feature is not necessary for  `either successful or coredump`,  just return error log message is ok.